code/modules/reagents/chemistry/reagents/drink_reagents.dm

```diff
@@ -661,9 +661,13 @@
 	desc = "Surprisingly it isn't grey."
 	icon_state = "grey_bull_glass"
 
-/datum/reagent/consumable/grey_bull/on_mob_metabolize(mob/living/affected_mob)
+/datum/reagent/consumable/grey_bull/on_mob_metabolize(mob/living/carbon/buller)
```

Comment:
Don't change parameter names, it breaks named parameters.

Reply:
If something in this proc would require the living mob to be using something defined on carbon, you'd have to redefine ``affected_mob`` to be a carbon after a ``iscarbon()`` check (for instance, to check for organs).

You'd need to do something like this.

```
if(iscarbon(affected_mob))
	var/mob/living/carbon/buller = affected_carbon
```

And then go from there when doing anything related to organ code and the like. I'm not sure if mood is present on the carbon level either, so maybe double check that. Otherwise, you might need to do an ``ishuman()`` check instead.

Reply:
I believe what Mothblocks is getting at is that the path change should be fine, just don't change the name.

Any proc that has `on_mob_metabolize(affected_mob = some_mob)` breaks on this change since the param name isn't the same across the proc's inheritance hierarchy.

So the PR author would want `mob/living/carbon/affected_mob` to keep the param name the same.

Reply:
Ah, okay, my b